packages/hoppscotch-backend/src/mailer/mailer.module.ts

```diff
@@ -46,3 +55,52 @@ export class MailerModule {
     };
   }
 }
+
+function isEnabled(value) {
+  return value === 'true';
+}
+function getMailerAddressFrom(env, config): string {
+  return (
+    env.INFRA.MAILER_ADDRESS_FROM ??
+    config.get('MAILER_ADDRESS_FROM') ??
+    throwErr(MAILER_SMTP_URL_UNDEFINED)
+  );
+}
+function getTransportOption(env, config): TransportType {
+  const useCustomConfigs = isEnabled(
+    env.INFRA.MAILER_USE_CUSTOM_CONFIGS ??
+      config.get('MAILER_USE_CUSTOM_CONFIGS'),
+  );
+
+  if (!useCustomConfigs) {
+    console.log('Using simple mailer configuration');
+    return (
+      env.INFRA.MAILER_SMTP_URL ??
+      config.get('MAILER_SMTP_URL') ??
+      throwErr(MAILER_SMTP_URL_UNDEFINED)
+    );
+  } else {
+    console.log('Using advanced mailer configuration');
+    return {
+      host: env.INFRA.MAILER_SMTP_HOST ?? config.get('MAILER_SMTP_HOST'),
+      port: +env.INFRA.MAILER_SMTP_PORT ?? +config.get('MAILER_SMTP_PORT'),
+      secure:
+        !!env.INFRA.MAILER_SMTP_SECURE ?? !!config.get('MAILER_SMTP_SECURE'),
+      auth: {
+        user:
+          env.INFRA.MAILER_SMTP_USER ??
+          config.get('MAILER_SMTP_USER') ??
+          throwErr(MAILER_SMTP_USER_UNDEFINED),
+        pass:
+          env.INFRA.MAILER_SMTP_PASSWORD ??
+          config.get('MAILER_SMTP_PASSWORD') ??
+          throwErr(MAILER_SMTP_PASSWORD_UNDEFINED),
+      },
+      tls: {
+        rejectUnauthorized:
+          !!env.INFRA.MAILER_TLS_REJECT_UNAUTHORIZED ??
+          !!config.get('MAILER_TLS_REJECT_UNAUTHORIZED'),
+      },
+    };
+  }
+}
```

Comment:
Move all the 3 helper functions into a new `helper.ts` file in the mailer module.